.github/workflows/test-build-spack.yaml

```diff
@@ -0,0 +1,80 @@
+name: Build ESMF Using Spack
+
+on:
+  workflow_dispatch:
+  push:
```

Comment:
I think `push` should be removed. It is too costly to run this action on every single push. This is more of a nightly test on `develop`, and of course available for manual dispatch.